### ISSUE TITLE
Regression regarding layout in the Site Editor

### ISSUE THREAD

**Issue by t-hamano** [WordPress/gutenberg#58044]
After #57938 was merged, there seem to be some layout breakdowns in the site editor. This issue summarizes the problems I discovered.

> [!NOTE]
> The following issue was discovered with the "New admin view" experimental flag **enabled**. There may be other issues if this experimental flag is turned off. One of them is attempted to be resolved in #58039.

### Templates are not displayed in the mobile layout

In fact, the expected screenshot itself doesn't seem right. I think the preview area itself needs to be hidden in the mobile layout.

| Expected | Actual (trunk) |
|--------|--------|
| ![image](https://github.com/WordPress/gutenberg/assets/54422211/a6b3dd2f-b2be-415c-af6c-b231756ad557) | ![image](https://github.com/WordPress/gutenberg/assets/54422211/a8f36083-2044-4115-98e0-0818d5ac7768)| 

### Canvas area and data view area overlap

You will see this issue when the viewport width is approximately between 800px and 1100px.

| Expected | Actual (trunk) |
|--------|--------|
| ![image](https://github.com/WordPress/gutenberg/assets/54422211/ca483547-c15c-44b6-8b56-7a3cc2d3b494) | ![image](https://github.com/WordPress/gutenberg/assets/54422211/abda44b9-6818-4330-a579-3cdefb114d40) |

<details><summary>Fixed</summary>

### The headers and the footer are not fixed on Templates, Template Parts, and Pattern pages

- #58077

| Expected | Actual (trunk) |
|--------|--------|
| ![image](https://github.com/WordPress/gutenberg/assets/54422211/8824b30e-f8b8-4009-a4f7-283f24f59e2b) | ![image](https://github.com/WordPress/gutenberg/assets/54422211/09872619-4334-4c16-9cb4-cfe4c9d7952e) |
| ![image](https://github.com/WordPress/gutenberg/assets/54422211/565d1651-f1fe-4816-bd36-55f1ad14e784) | ![image](https://github.com/WordPress/gutenberg/assets/54422211/5c85ed04-4239-4eb3-a670-95ab0b8039a3) | 

### Top right and bottom right corners are not rounded

- #58077

Presumably, this can only be reproduced in the Chrome browser on Windows OS. It happens on All Templates, Template Parts, and Pattern pages.

| Expected | Actual (trunk) |
|--------|--------|
| ![expected](https://github.com/WordPress/gutenberg/assets/54422211/80884af5-4cd6-472f-8756-c938a866d8ab) | ![actual](https://github.com/WordPress/gutenberg/assets/54422211/5f73b753-13c5-4ef9-845e-a0cfc24f2cac) | 

### Extra space on top and right side

- #58077

| Expected | Actual (trunk) |
|--------|--------|
| ![image](https://github.com/WordPress/gutenberg/assets/54422211/014e67ff-ab1e-4fad-8825-d2bb0d581997) | ![image](https://github.com/WordPress/gutenberg/assets/54422211/76e5d942-b505-4c71-9a71-042473102d8a)

### Pattern padding

https://github.com/WordPress/gutenberg/pull/58139 stabilizes the Patterns page with DataViews, so this issue effectively no longer needs to be fixed.

Disable the data views experiment to see this:

<img src="https://private-user-images.githubusercontent.com/846565/298613458-0b7a6765-bb41-4d1f-bc74-9ca887eb9e12.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDYwMjU3MTIsIm5iZiI6MTcwNjAyNTQxMiwicGF0aCI6Ii84NDY1NjUvMjk4NjEzNDU4LTBiN2E2NzY1LWJiNDEtNGQxZi1iYzc0LTljYTg4N2ViOWUxMi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMTIzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDEyM1QxNTU2NTJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mZjczMjMyNGE4ODhjMjgyYmQyMGExMDBkNjc5NmIwYWNmMTUzODJmMzhiMmZlOTM1MjhmMDk2YjYyNzFkYzZlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.4khxx2Y0gM69J1-vhwbsu3Z8iL64ZdK8NVWcN93KwiE">

With the margin added to `.edit-site-layout__area` the vertical stroke on the left doesn't quite reach the top/bottom of the viewport, and the pattern thumbnails are prematurely clipped.

</details>

**Comment by youknowriad** [comment]
Thanks for flagging these regressions. Any help addressing some of them would be appreciated.

**Comment by jameskoster** [comment]
Another issue in the Pattern management view with dataview experiments disabled:

<img width="1499" alt="Screenshot 2024-01-22 at 14 13 38" src="https://github.com/WordPress/gutenberg/assets/846565/0b7a6765-bb41-4d1f-bc74-9ca887eb9e12">

With the margin added to `.edit-site-layout__area` the vertical stroke on the left doesn't quite reach the top/bottom of the viewport, and the pattern thumbnails are prematurely clipped.

**Comment by colorful-tones** [comment]
@t-hamano have all the regressions noted been addressed in the merged PRs? If so, can we move this to ‘Done’? Or is further testing required?

**Comment by t-hamano** [comment]
> have all the regressions noted been addressed in the merged PRs?

No, although resolved issues are grouped under the "Fixed" accordion, the following issues still remain.

- Templates are not displayed in the mobile layout
- Canvas area and data view area overlap
- ~~Pattern padding~~

**Comment by t-hamano** [comment]
> ### Pattern padding
> 
> Disable the data views experiment to see this:
> 
> With the margin added to `.edit-site-layout__area` the vertical stroke on the left doesn't quite reach the top/bottom of the viewport, and the pattern thumbnails are prematurely clipped.

Update: #58139 stabilizes the Patterns page with DataViews, so this issue effectively no longer needs to be fixed.

**Comment by t-hamano** [comment]
The issues that currently remain are regarding the responsive layout and the mobile viewport. Especially when it comes to the mobile layout, there are two serious issues:

- On the "Patterns" page, nothing happens when you click on a pattern category
- Data view is not displayed on "Manage all pages" page

My understanding is that the data view has been stabilized on all pages and shipped in WP.6.5. I believe this issue should be fixed in WP6.5, so I would like to label this issue as a high priority.

https://github.com/WordPress/gutenberg/assets/54422211/8af5d38e-c7cc-47d6-ac1f-9cb28adf7139

**Comment by annezazu** [comment]
Might this be related or does this need a new issue? Noticed it today using WP playground on 6.5-alpha-57531



https://github.com/WordPress/gutenberg/assets/26996883/c3ce1969-0eaf-47a4-b4a0-5801ab47ac53

**Comment by t-hamano** [comment]
> Might this be related or does this need a new issue? Noticed it today using WP playground on 6.5-alpha-57531

This seems to occur when the page slug is extremely long. I was able to see this problem in WP6.4.3 as well, so I don't think it's related to this issue. Therefore, I think it would be better to submit a new issue.

**Comment by annezazu** [comment]
Done :) https://github.com/WordPress/gutenberg/issues/58748

**Comment by colorful-tones** [comment]
I have had zero testing with the new admin views and thought this would be a great way to get started 🙃 . I'm unclear about the testing instructions, but I detailed what I tested below and hopefully it helps.

## Templates are not displayed in the mobile layout.

I tested this against the following scenarios and obtained the same experience on all of them.

- Gutenberg 19622fa6157729bcef21cb5c8606d9dd1d9f9dc6 | Gutenberg > Experiments > New Admin Views: **On** | WordPress **6.5-alpha-57571** (nightly)
- Gutenberg 19622fa6157729bcef21cb5c8606d9dd1d9f9dc6 | Gutenberg > Experiments > New Admin Views: **Off** | WordPress **6.5-alpha-57571** (nightly)
- Gutenberg 19622fa6157729bcef21cb5c8606d9dd1d9f9dc6 | Gutenberg > Experiments > New Admin Views: **On** | WordPress **6.4.3**
- Gutenberg 19622fa6157729bcef21cb5c8606d9dd1d9f9dc6 | Gutenberg > Experiments > New Admin Views: **Off** | WordPress **6.4.3**

This 👇  is what I saw for all of those 👆  scenarios

![Site Editor > Templates view](https://github.com/WordPress/gutenberg/assets/405912/961107d1-6206-4415-bac8-fe1011e7d206)

## Canvas area and data view area overlap

With viewport width at approximately between 800px and 1100px.

I tested this against the following scenarios and obtained the same experience on all of them. 

- Gutenberg 19622fa6157729bcef21cb5c8606d9dd1d9f9dc6 | Gutenberg > Experiments > New Admin Views: **On** | WordPress **6.5-alpha-57571** (nightly)
- Gutenberg 19622fa6157729bcef21cb5c8606d9dd1d9f9dc6 | Gutenberg > Experiments > New Admin Views: **Off** | WordPress **6.5-alpha-57571** (nightly)
- Gutenberg 19622fa6157729bcef21cb5c8606d9dd1d9f9dc6 | Gutenberg > Experiments > New Admin Views: **On** | WordPress **6.4.3**
- Gutenberg 19622fa6157729bcef21cb5c8606d9dd1d9f9dc6 | Gutenberg > Experiments > New Admin Views: **Off** | WordPress **6.4.3**

This 👇  is what I saw for all of those 👆  scenarios

https://github.com/WordPress/gutenberg/assets/405912/3d5ec722-38ae-4409-9b4a-1ecfb996858d

## On the "Patterns" page, nothing happens when you click on a pattern category

I tested this against the following scenarios and obtained the same experience on all of them. 

- Gutenberg 19622fa6157729bcef21cb5c8606d9dd1d9f9dc6 | Gutenberg > Experiments > New Admin Views: **On** | WordPress **6.5-alpha-57571** (nightly)
- Gutenberg 19622fa6157729bcef21cb5c8606d9dd1d9f9dc6 | Gutenberg > Experiments > New Admin Views: **Off** | WordPress **6.5-alpha-57571** (nightly)
- Gutenberg 19622fa6157729bcef21cb5c8606d9dd1d9f9dc6 | Gutenberg > Experiments > New Admin Views: **On** | WordPress **6.4.3**
- Gutenberg 19622fa6157729bcef21cb5c8606d9dd1d9f9dc6 | Gutenberg > Experiments > New Admin Views: **Off** | WordPress **6.4.3**

This 👇  is what I saw for all of those 👆  scenarios. **I confirmed that clicking on _any_ of the pattern categories did nothing for all of the above scenarios.** 🐛 

![Site Editor > Patterns screen](https://github.com/WordPress/gutenberg/assets/405912/feec7c37-a003-4929-a7d1-935d4b77512b)

## Data view is not displayed on "Manage all pages" page

I tested this against the following scenarios:

- Gutenberg 19622fa6157729bcef21cb5c8606d9dd1d9f9dc6 | Gutenberg > Experiments > New Admin Views: **On** | WordPress **6.5-alpha-57571** (nightly)
- Gutenberg 19622fa6157729bcef21cb5c8606d9dd1d9f9dc6 | Gutenberg > Experiments > New Admin Views: **On** | WordPress **6.4.3**

This 👇  is what I saw for these two 👆  scenarios.

https://github.com/WordPress/gutenberg/assets/405912/a9245a27-af5b-457e-9a3d-5ce257d6cc56

I also tested against the following scenarios:

- Gutenberg 19622fa6157729bcef21cb5c8606d9dd1d9f9dc6 | Gutenberg > Experiments > New Admin Views: **Off** | WordPress **6.5-alpha-57571** (nightly)
- Gutenberg 19622fa6157729bcef21cb5c8606d9dd1d9f9dc6 | Gutenberg > Experiments > New Admin Views: **Off** | WordPress **6.4.3**

This 👇  is what I saw for these two 👆  scenarios.

![Site Editor > Pages > Manage All Pages screen](https://github.com/WordPress/gutenberg/assets/405912/0a7b30eb-d921-4e57-b136-cdd53cfab9b9)
